src/UI/Balance/Balance.tsx

```diff
@@ -35,33 +35,68 @@ export const BalanceComponent = ({
       ? `$${mainBalance}`
       : mainBalance;
 
+  const getBalancePayload = () => {
+    const balancePayload: Record<string, string> = {};
+
+    if (addEqualSign && displayAsUsd) {
+      Object.assign(balancePayload, { approximation: '≈' });
+    }
+
+    if (processedMainBalance) {
+      Object.assign(balancePayload, { processedMainBalance });
+    }
+
+    if (decimalBalance) {
+      Object.assign(balancePayload, { decimalBalance: `.${decimalBalance}` });
+    }
+
+    if (!displayAsUsd && showTokenLabel) {
+      Object.assign(balancePayload, { tokenLabel: ` ${tokenLabel}` });
+    }
+
+    return balancePayload;
+  };
+
+  const balancePayload = getBalancePayload();
+  const dataBalanceValue = Object.values(balancePayload).reduce(
```

Comment:
I would not agree with this because you have no guarantee of object proprety order. And the order is not human readable

Reply:
The order is given by the order of the conditions in the `getBalancePayload` function.